types/datum.go

```diff
@@ -1444,6 +1444,9 @@ func (d *Datum) ToFloat64(sc *stmtctx.StatementContext) (float64, error) {
 	case KindBinaryLiteral, KindMysqlBit:
 		val, err := d.GetBinaryLiteral().ToInt()
 		return float64(val), err
+	case KindMysqlJSON:
+		f, err := ConvertJSONToFloat(sc, d.GetMysqlJSON())
+		return f, err
```

Comment:
We should return `f, errors.Trace(err)` to track the whole error stack. You can modify lIne 1446 together.